reframe/core/config.py

```diff
@@ -67,6 +66,11 @@ def __init__(self, site_config, filename):
         self._subconfigs = {}
         self._local_system = None
         self._sticky_options = {}
+        self.autodetect_opts = {
```

Comment:
It's a small thing, but since you are not using this outside the `_SiteConfig` object I would name it `self._autodetect_opts`. Fine if you decide to leave it as it.